cmd/fossa/cmd/test/test_test.go

```diff
@@ -0,0 +1,169 @@
+package test
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/fossas/fossa-cli/api/fossa"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+)
+
+const orgID = 3
+
+// taskStatus is a struct that imitates the anonymous struct within fossa.Build
+type taskStatus struct {
+	Status string
+}
+
+func TestRunCustomFetcherCustomProject(t *testing.T) {
+	c := make(chan string)
+	locator := "custom+" + strconv.Itoa(orgID) + "%2FtestRun$1000"
+	ts := testServer(locator, c)
+
+	flagSet := testFlags("custom", "testRun", ts.URL, 1000)
+	context := cli.NewContext(&cli.App{}, flagSet, &cli.Context{})
+
+	go Run(context)
+	msg := <-c
+	if msg != "SUCCESS" {
+		assert.Equal(t, locator, msg)
+	}
+
+	ts.Close()
+}
+
+func TestRunCustomFetcherGitSSHProject(t *testing.T) {
```

Comment:
This looks like a good fit for table-driven tests. See https://github.com/golang/go/wiki/TableDrivenTests.